nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -356,7 +356,12 @@ Costmap2DROS::mapUpdateLoop(double frequency)
 
     // Measure the execution time of the updateMap method
     timer.start();
-    updateMap();
+    try {
+      updateMap();
+    }
+    catch (const char* e) {
+      RCLCPP_ERROR(get_logger(), e);
```

Comment:
This should never happen since we have the `map_received` flag now. I'd

1.    not bother to catch the exception,
2.    change this to RCLCPP_FATAL, or
3.   rethrow after printing some helpful message.

My vote is 1.

Reply:
Agreed on #1

Reply:
Done - not catching